File: leetjava/two_sum.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        // should return indecies that add up to the target  
        // using HashMap
        // declare Hashmap
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        // declare the answer
        int[] result = new int[2];
        // then loop through the numbers and add them to the result
        for(int i = 0; i < numbers.length; i++){
            // map all the numbers in two sets of results
            if(map.containsKey(numbers[i])){
                int index = map.get(numbers[i]);
                result[0]= index + 1; 
                result[1]= i+1; 
                break;
            }else{
                map.put(target-numbers[i], i);
            }
        }
        return result;
    }
}

File: leetcode/two_sum.py
def twoSum(self, nums, target):
    # dictionary for numbers that are visited 
    seen = {}
    # declare visited, i and remaining
    # then enumate through the nums and mark i and visited
    for i, visited in enumerate(nums):
        # remaining is the target minus the visited due to visited plus i should target
        remaining = target - visited
            # then add them all the new dictionary 
        if remaining in seen:
            return [seen[remaining], i]
        # mark all the nums with i
        seen[visited] = i
    # then return the dict
    return []
# self = 0
target = 9 
nums = [2,7,11,15]
print(twoSum(0, nums=nums, target=target))